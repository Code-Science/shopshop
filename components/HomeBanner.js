import React from 'react';
import Image from 'next/image';
import Link from 'next/link';
import styles from '../styles/components/HomeBanner.module.css';

function HomeBanner() {
  return (
    <div className={styles.homeBanner}>
      <ul className={styles.images}>
        <li className={styles.image}>
          <Image
            src="/images/homebanner1.jpeg"
            alt="home banner"
            objectFit="cover"
            layout="fill"
          />
        </li>
        <li className={styles.image}>
          <Image
            src="/images/homebanner2.jpg"
            alt="home banner"
            objectFit="cover"
            layout="fill"
          />
        </li>
        <li className={styles.image}>
          <Image
            src="/images/homebanner3.jpg"
            alt="home banner"
            objectFit="cover"
            layout="fill"
          />
        </li>
        <li className={styles.image}>
          <Image
            src="/images/homebanner4.jpg"
            alt="home banner"
            objectFit="cover"
            layout="fill"
          />
        </li>
        <li className={styles.image}>
          <Image
            src="/images/homebanner5.jpg"
            alt="home banner"
            objectFit="cover"
            layout="fill"
          />
        </li>
      </ul>
      <div className={styles.shade}>
        <ul className={styles.textBox}>
          <li className={styles.text}>
            MODERN Fashion Craze{' '}
            <Link href="/collection/women/accessories">
              <a className={styles.shopBtn}>
                <br />
                Shop now
              </a>
            </Link>
          </li>
          <li className={styles.text}>
            ELEGENT New Collection{' '}
            <Link href="/collection/women/accessories">
              <a className={styles.shopBtn}>
                <br />
                Shop now
              </a>
            </Link>
          </li>
          <li className={styles.text}>
            WINTER New Season{' '}
            <Link href="/collection/women/clothes">
              <a className={styles.shopBtn}>
                <br />
                Shop now
              </a>
            </Link>
          </li>
          <li className={styles.text}>
            MENSWEAR New Collection{' '}
            <Link href="/collection/men/accessories">
              <a className={styles.shopBtn}>
                <br />
                Shop now
              </a>
            </Link>
          </li>
          <li className={styles.text}>
            Kids New Designs{' '}
            <Link href="/collection/kids/clothes">
              <a className={styles.shopBtn}>
                <br />
                Shop now
              </a>
            </Link>
          </li>
        </ul>
      </div>
    </div>
  );
}

export default HomeBanner;
