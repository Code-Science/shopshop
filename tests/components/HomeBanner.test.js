import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import HomeBanner from '../../components/HomeBanner';

jest.mock('../../store/StateProvider');

configure({ adapter: new Adapter() });

describe('HomeBanner Component', () => {
  const wrapper = shallow(<HomeBanner />);

  it('expect to render HomeBanner component', () => {
    expect(wrapper.length).toEqual(1);
  });
  it('snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
