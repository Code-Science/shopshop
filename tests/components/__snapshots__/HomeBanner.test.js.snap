// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeBanner Component snapshot test 1`] = `
<div
  className="homeBanner"
>
  <ul
    className="images"
  >
    <li
      className="image"
    >
      <Image
        alt="home banner"
        layout="fill"
        objectFit="cover"
        src="/images/homebanner1.jpeg"
      />
    </li>
    <li
      className="image"
    >
      <Image
        alt="home banner"
        layout="fill"
        objectFit="cover"
        src="/images/homebanner2.jpg"
      />
    </li>
    <li
      className="image"
    >
      <Image
        alt="home banner"
        layout="fill"
        objectFit="cover"
        src="/images/homebanner3.jpg"
      />
    </li>
    <li
      className="image"
    >
      <Image
        alt="home banner"
        layout="fill"
        objectFit="cover"
        src="/images/homebanner4.jpg"
      />
    </li>
    <li
      className="image"
    >
      <Image
        alt="home banner"
        layout="fill"
        objectFit="cover"
        src="/images/homebanner5.jpg"
      />
    </li>
  </ul>
  <div
    className="shade"
  >
    <ul
      className="textBox"
    >
      <li
        className="text"
      >
        MODERN Fashion Craze
         
        <Link
          href="/collection/women/accessories"
        >
          <a
            className="shopBtn"
          >
            <br />
            Shop now
          </a>
        </Link>
      </li>
      <li
        className="text"
      >
        ELEGENT New Collection
         
        <Link
          href="/collection/women/accessories"
        >
          <a
            className="shopBtn"
          >
            <br />
            Shop now
          </a>
        </Link>
      </li>
      <li
        className="text"
      >
        WINTER New Season
         
        <Link
          href="/collection/women/clothes"
        >
          <a
            className="shopBtn"
          >
            <br />
            Shop now
          </a>
        </Link>
      </li>
      <li
        className="text"
      >
        MENSWEAR New Collection
         
        <Link
          href="/collection/men/accessories"
        >
          <a
            className="shopBtn"
          >
            <br />
            Shop now
          </a>
        </Link>
      </li>
      <li
        className="text"
      >
        Kids New Designs
         
        <Link
          href="/collection/kids/clothes"
        >
          <a
            className="shopBtn"
          >
            <br />
            Shop now
          </a>
        </Link>
      </li>
    </ul>
  </div>
</div>
`;
